packages/app/src/common/hooks/useAddressLookup.ts

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from "react";
+import { useConnection } from "@solana/wallet-adapter-react";
+import { walletNameToAddressAndProfilePicture } from "@portal-payments/solana-wallet-names";
+
+const useAddressLookup = (inputProp: string): any => {
+  const { connection } = useConnection();
+  const [walletDomainName, setWalletDomainName] = useState("");
+  const [resolvedDomain, setResolvedDomain] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const [output, setOutput] = useState<{
+    resolvedDomain: string | null;
+    errorMessage: string | null;
+  }>({ resolvedDomain: null, errorMessage: null });
+
+  // Extracting wallet domain name from wallet address
+  const extractDomainName = (input: string): string | null => {
+    if (input.includes(".")) {
+      const parts = input.split(".");
+      return parts[parts.length - 2] + "." + parts[parts.length - 1];
+    } else return null;
+  };
+
+  useEffect(() => {
```

Comment:
you can probably simplify these a little - rather than having three useEffects, you can probably just have one, which calls walletNameToAddressAndProfilePicture if the inputProp is a domain. I don't think you need to store the intermediary results walletDomainName and resolvedDomain in state hooks.